course7/homeworks/patrickkostjens/complex/main.cc

```diff
@@ -0,0 +1,44 @@
+#include "calculator.hh"
+
+#include <boost/filesystem.hpp>
+#include <string>
+#include <dlfcn.h>
+
+namespace fs = boost::filesystem;
+
+int main(int argc, char** argv)
+{
+	Calculator c;
+	int plugin_count = 0;
+	for (const auto& it : fs::directory_iterator(fs::path(".")))
+		if (it.path().extension() == ".so" || it.path().extension() == ".dylib")
+			plugin_count++;
+
+	plugin *plugins = new plugin[plugin_count];
```

Comment:
The "new" again...
Never use it, unless you are working on some low-level stuff. And then, if you do, every "new" should be paired with "delete".
If you need variable-sized storage, use ```std::vector```

Reply:
Removed. I'm still getting used to `new` not being needed basically.